compiler/packages/babel-plugin-react-compiler/src/HIR/BuildHIR.ts

```diff
@@ -3545,7 +3545,7 @@ function lowerAssignment(
   builder: HIRBuilder,
   loc: SourceLocation,
   kind: InstructionKind,
-  lvaluePath: NodePath<t.LVal>,
+  lvaluePath: NodePath<t.LVal | t.OptionalMemberExpression>,
```

Comment:
why? this PR shouldn't touch the compiler internals at all

Reply:
After deduping babel types, the `AssignmentExpression` type's `left` property includes `OptionalMemberExpression` now. 
![image](https://github.com/user-attachments/assets/3af653d3-234b-4a51-a482-2e66b0920c16)

![image](https://github.com/user-attachments/assets/51e46531-ae67-4dbc-9eae-880e6f000b2b)
![image](https://github.com/user-attachments/assets/30856b58-4e62-46b3-a4f8-ca099aecb704)

Reply:
I wonder if this accidentally leaked from their v8 pre-release.  It looks like it was originally released in v8 alpha https://github.com/babel/babel/commit/52244bf0834801684f1e55621b395f5af69f6a10#diff-8686732cac4b4c0405e0415ac800fb066039f1ae83786248e2e0c50924f17adf

But it's in 7.26 now

Reply:
This seems to still only be Stage 1: 
- https://github.com/tc39/proposal-optional-chaining-assignment 
- https://github.com/babel/babel/pull/15751

I don't think we should support this in the compiler, as it would require more changes than just a simple type update. We should throw instead on the call site(s) to `lowerAssignment` if the type is of `OptionalMemberExpression`.

Reply:
Updated to throw at call site.